src/example/ERC20TransferModule.sol

```
@@ -0,0 +1,111 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.20;
+
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import {ContractModuleBase} from "../core/ContractModuleBase.sol";
+import {CrossContext} from "../core/IContractModule.sol";
+
+abstract contract ERC20TransferModule is ContractModuleBase {
+    using SafeERC20 for IERC20;
```

Question: What is the reason we need to use SafeERC20?

---

I think
- Compatibility: Handles non-standard tokens (like USDT) that return nothing, avoiding reverts.
- Safety: Forces a revert if a transfer fails, preventing silent failures.